frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1049,7 +1052,9 @@ export default ({
             $state.complete()
             if (!this.$refs.conversation ||
             this.$refs.conversation.scrollHeight === this.$refs.conversation.clientHeight) {
-              const msg = this.messages.filter(m => !m.pending).slice(-1)
+              // updateReadUntilMessageHash should only use messages that exist
+              // in the contract
+              const msg = this.messages.filter(m => !m.pending && !m.hasFailed).pop()
```

Comment:
As above, `.pop()` doesn't mutate `this.messages` because `.filter()` returns a new array. I think this check here might not be needed as this is mostly processing new events, which should all have incoming direction. However, I decided to keep it to be safe (as the chatroom breaks badly if non-existing messages are added) as this is still operating `this.messages`, which could have previous contents.